libs/dh/metering-point/feature-tabs/src/lib/processes-tab-content/dh-processes-table/dh-processes-table.component.ts

```diff
@@ -49,66 +51,132 @@ import { TranslocoModule } from '@ngneat/transloco';
 })
 export class DhProcessesTableComponent implements AfterViewInit {
   displayedColumns: string[] = [
-    'expand',
+    'tableExpandControl',
     'name',
     'createdDate',
     'effectiveDate',
     'status',
     'hasDetailsErrors',
   ];
   iconSize = WattIconSize;
-  sortedData: DhProcess[] = [];
-  @Input()
-  processes: DhProcess[] = [];
+  sortedData: DhProcessesTableRow[] = [];
+
+  @Input() processes: DhProcess[] = [];
+
+  processRows: DhProcessesTableRow[] = [];
 
   @ViewChild(MatSort) matSort?: MatSort;
 
+  private static wrapInTableRow(data: DhProcess[]): DhProcessesTableRow[] {
+    return data.map((process) => ({
+      data: process,
+      expanded: false,
+      height: 0,
+    }));
+  }
+
+  private static getRowToExpand(
```

Comment:
I'm not sure I follow the logic here - Are you saying that private methods should never be static?

Reply:
> Are you saying that private methods should never be static?

No :)

Maybe I miss something about static methods but aren't they useful because they can be called without instantiating a class? I this case we're inside the class so I'm not sure marking a method as static adds any benefits. So we might as well remove the `static` keyword and just call it with `this.wrapInTableRow(...)`.

Reply:
Usually static is a good way to indicate that a function is pure and thereby doesn't change the state of the object - A bit of functional programming if you will.

Reply:
I see. Well, what I'd usually would do in that case then is to move the method/function out of the class to its own file. This will make it more obvious that the function is a kind of utility and can be tested on its own with unit tests.

Up to you if you want to do that or leave the method in the class. I pointed it because we don't have that many `static` methods in the code base.